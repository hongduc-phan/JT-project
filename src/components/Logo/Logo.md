Normal (Get the first alphabet of the company)

```js
const Logo = require('./').default;

<Logo company="Expresso Media" />;
```

With Logo

```js
const Logo = require('./').default;

<Logo
  company="JuzTalent"
  logo="https://juztalent.com/wp-content/uploads/2017/01/cropped-juztalent.png"
/>;
```
