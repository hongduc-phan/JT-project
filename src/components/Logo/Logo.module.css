.logo {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 2.25rem;
  height: 2.25rem;
  border-radius: 50%;
}

.logoWithIcon {
  width: unset;
  border-radius: 0;
}

.logoDefault {
  font-weight: var(--font-weight-semibold);
  line-height: 1.33;
  background-color: #84cdf0;
}
