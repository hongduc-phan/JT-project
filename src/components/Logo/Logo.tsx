import React, {AllHTMLAttributes} from 'react';
import clsx from 'clsx';

import Typo, {TypoVariants, TypoColors, TypoAlignment} from '../Typo';

import styles from './Logo.module.css';

export interface ILogoProps extends AllHTMLAttributes<HTMLDivElement> {
  logo?: string;
  company: string;
  className?: string;
}

function getFirstCharacter(text: string) {
  if (!text) {
    return '';
  }
  return text[0].toLocaleUpperCase();
}

const Logo = ({logo, company, className, ...others}: ILogoProps) => {
  return (
    <div className={className} {...others}>
      {logo ? (
        <img
          className={clsx(styles.logo, styles.logoWithIcon)}
          src={logo}
          alt={company}
        />
      ) : (
        <Typo
          variant={TypoVariants.h4}
          color={TypoColors.white}
          align={TypoAlignment.middle}
          className={clsx(styles.logo, styles.logoDefault)}
        >
          {getFirstCharacter(company)}
        </Typo>
      )}
    </div>
  );
};

export default Logo;
