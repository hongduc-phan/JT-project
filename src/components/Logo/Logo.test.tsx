import React from 'react';
import renderer from 'react-test-renderer';
import {mount, ReactWrapper} from 'enzyme';

import Logo from './Logo';

describe('<Logo />', () => {
  describe('Normal', () => {
    let normalLogo: any;
    let component: ReactWrapper;

    beforeEach(() => {
      normalLogo = <Logo company="JuzTalent" />;

      component = mount(normalLogo);
    });

    it('Should alway render <span> tag', () => {
      const spans = component.find('span');
      expect(spans.length).toBeGreaterThan(0);
    });

    it('Should generate the first alphabel of company name', () => {
      expect(component.find('span').text()).toEqual('J');
    });

    it('Should match logo snapshot', () => {
      const tree = renderer.create(normalLogo).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('With logo', () => {
    let normalLogo: any;
    let component: ReactWrapper;

    beforeEach(() => {
      normalLogo = (
        <Logo
          logo="https://juztalent.com/wp-content/uploads/2017/01/cropped-juztalent.png"
          company="JuzTalent"
        />
      );

      component = mount(normalLogo);
    });

    it('Should alway render <img> tag', () => {
      const imgs = component.find('img');
      expect(imgs.length).toBeGreaterThan(0);
    });

    it('Should match logo with logo snapshot', () => {
      const tree = renderer.create(normalLogo).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
