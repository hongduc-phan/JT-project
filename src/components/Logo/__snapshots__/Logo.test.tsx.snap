// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> Normal Should match logo snapshot 1`] = `
<div>
  <span
    className="typo logo logoDefault h4 white middle"
  >
    J
  </span>
</div>
`;

exports[`<Logo /> With logo Should match logo with logo snapshot 1`] = `
<div>
  <img
    alt="JuzTalent"
    className="logo logoWithIcon"
    src="https://juztalent.com/wp-content/uploads/2017/01/cropped-juztalent.png"
  />
</div>
`;
